src/main/java/com/clouway/cuse/spi/SearchEngineImpl.java

```diff
@@ -56,4 +71,16 @@ public void delete(String indexName, List<Long> objectIds) {
     indexRegistry.delete(indexName, objectIds);
   }
 
+  private boolean HomogeneousListChecker(Object matcher, List<? extends Object> objectsToMatch) {
+    Field[] matcherFields = matcher.getClass().getDeclaredFields();
+    for (Object object : objectsToMatch) {
+      Field[] objectFields = object.getClass().getDeclaredFields();
```

Comment:
I think we could check the class names instead of their fields.
also the best validation could be checking the @SearchIndex and the name of the index inside
something like that
```java
((SearchIndex)clazz.getDeclaredAnnotations()[0]).name()
```